### PR TITLE
Ionic/integration

### LINKED ISSUE #2454: Infinite Scrolling at History

### LINKED ISSUE #3924: UI Idea: Option to hide balance from main screen
My balance is the most prominent feature on the main screen. I may not want someone looking over my shoulder to see my balance and see extremely wealthy I am. (I wish...)

It would be nice to have the option to hide the balance for privacy reasons.

### PR DESCRIPTION
**Pull request by JDonadio**
Features : 

- Tap to lock balance . Fixes #3924 
- Balance pull to refresh
- Transactions infinite scroll. Fixes #2454 
- More responsive modals and popups
- Draggable side bar
- Add comments to transactions

### CODE REVIEW

**Comment by matiu** [comment]
needs rebase

**Comment by matiu** [comment]
reviewing.

**Comment by matiu** [comment]
Fantastic job guys!!

Some initial comments:

 - Backup view empty, if there is not seed
 - Numbers for txs / total should be shown at the end of the listing, something like:
     
           \/ 432 more transactions

           \/ 432 more

- Icons are broken in Chrome app
- When the comment is edited, the modal should be titled:
     'Edit comment'
and the form should be completed with the  current comment.
- Remove the "Save an empty content to delete it"
- Search should look also on notes.  When a note is edited, the 'search cache' should be reset


```

**Comment by matiu** [comment]
Tested on Chrome App + OSX NW.

File: .gitignore [hunk 1]
```diff
@@ -24,7 +24,8 @@ webkitbuilds/*
 !webkitbuilds/build-osx.sh
 !webkitbuilds/Background.png
 
-
+#fonts
+public/fonts
```
Comment (matiu) [inline .gitignore]: is this installed by a npm package? if not, why is ignored?

File: public/index.html [hunk 2]
```diff
@@ -6,46 +6,48 @@
...
-              ng-class="{
-                'main': index.hasProfile
-              }"></section>
-          </div>
-        </div>
-        <a class="close-menu" ng-click="index.closeMenu()"></a>
+  <ion-side-menus class="page" ng-controller="indexController as index">
+
+    <!-- Left menu -->
+    <ion-side-menu side="left">
```
Comment (matiu) [inline public/index.html]: much cleaner 👍

File: public/views/modals/tx-details.html [hunk 3]
```diff
@@ -1,148 +1,168 @@
...
+
+      <div ng-show="btx.txid" class="tx-details-blockchain">
+        <div class="text-center m20t">
+          <button class="button outline round dark-gray tiny" ng-click="$root.openExternalLink('https://' +
+            (getShortNetworkName() == 'test' ? 'test-' : '') + 'insight.bitpay.com/tx/' + btx.txid)">
+            <span class="text-gray" translate>See it on the blockchain</span>
+          </button>
+          <button class="button outline round dark-gray tiny" ng-click="showCommentPopup(comment)">
+            <span class="text-gray" translate ng-show="!comment">Add a comment <i class="ion-ios-plus-outline"></i></span>
+            <span class="text-gray" translate ng-show="comment">Edit comment <i class="ion-ios-compose-outline"></i></span>
```
Comment (matiu) [inline public/views/modals/tx-details.html]: Please remove the icons from the buttons.

File: src/js/controllers/create.js [hunk 4]
```diff
@@ -91,6 +89,7 @@ angular.module('copayApp.controllers').controller('createController',
     this.create = function(form) {
       if (form && form.$invalid) {
         this.error = gettext('Please enter the required fields');
+        $ionicScrollDelegate.scrollTop();
```
Comment (matiu) [inline src/js/controllers/create.js]: Interesting.

File: .gitignore (hunk 1 above)
Comment (gabrielbazan7) [inline .gitignore]: It is installed together with ionic by bower.

**Comment by matiu** [comment]
TODO:
 - [ ] make the TX comments searchable
 - [ ] export TX comments on the CSV

**Comment by Kirvx** [comment]
"Confirmer" is truncated 

sans titre

**Comment by JDonadio** [comment]
Thanks for comment @Kirvx, we already made the fix ;)

File: public/views/includes/confirm-tx.html [hunk 5]
```diff
@@ -1,47 +1,39 @@
...
-      <div class="large-6 medium-6 small-6 columns">
-        <button 
-           ng-click="confirm.accept(index.confirmTx.callback)"
-           class="small m10b round expand" 
-           ng-style="{'background-color':index.backgroundColor}" translate>
-          Confirm
-        </button>
-      </div> 
+    <div class="half-row left">
+      <button ng-click="accept()" class="round expand" ng-style="{'background-color':index.backgroundColor}" translate>
```
Comment (Kirvx) [inline public/views/includes/confirm-tx.html]: translate should be deleted here.

Comment (gabrielbazan7) [inline public/views/includes/confirm-tx.html]: Thanks

**Comment by matiu** [comment]
Great work!! @JDonadio and @gabrielbazan7, and also @ajp8164 !

**Comment by Kirvx** [comment]
@JDonadio Same problem of truncated strings with "Enter a new comment" for "Enregistrer"
sans titre

**Pull request merged**